### DISCUSSION TITLE
Unexpected type PyCharm tooltip

### DISCUSSION THREAD

**Discussion by Arbiter-S** [sqlalchemy/sqlalchemy#11070] [Usage Questions] [answered]
I'm making a telegram bot script using [python-telegram-bot](https://github.com/python-telegram-bot/python-telegram-bot) and use SQLAlchemy for as my ORM.
I wrote this statement which which gives me unexpected type tooltip on PyCharm on the where clause.
`statement = select(Account).where(Account.email == context.user_data['email'])`

The tooltip:
Expected type 'ColumnElement[bool] | _HasClauseElement[bool] | SQLCoreOperations[bool] | ExpressionElementRole[bool] | () -> ColumnElement[bool] | LambdaElement', got 'bool' instead

Account model:
```
class Account(Base):
    __tablename__ = 'account'

    id = mapped_column(Integer, primary_key=True, autoincrement=True)
    email = mapped_column(String)
```

The code works with no issues but I don't know what is causing this.

**Comment by CaselIT** [comment]
Hi,

This seems a pycharm bug. Probably not something sqlalchemy can change or solve 

Btw if you expect typing to work correctly you will need to type your models properly using mapped etc.

**Comment by Klikini** [comment, marked as answer]
This is a known PyCharm bug and they're actively working on it: https://youtrack.jetbrains.com/issue/PY-59791.

**Reply by CaselIT** [reply]
thanks for reporting it
